uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -246,4 +246,42 @@ public static function getMostRecentStartDateAvailable()
 
     return date('Y-m-d');
   }
+
+  /**
+   * Returns the number of working days for this Absence Period.
+   *
+   * The number is given by "number of ways in period" - "weekends" - "public holidays".
+   *
+   * If a public holiday falls on a weekend, we only count one day.
+   *
+   * @return int
+   */
+  public function getNumberOfWorkingDays()
+  {
+    $startDate = new DateTime($this->start_date);
+    $endDate = new DateTime($this->end_date);
+    $oneDayInterval = new DateInterval('P1D');
+
+    // DatePeriod doesn't include the end date,
+    // so we add one more day for it to be included
+    $endDate->add($oneDayInterval);
+
+    $numberOfWorkingDays = 0;
+    $period = new DatePeriod($startDate, $oneDayInterval, $endDate);
+    foreach($period as $date) {
+      $dayOfTheWeek = $date->format('N');
+      $dayIsWorkingDay = $dayOfTheWeek > 0 && $dayOfTheWeek < 6;
```

Comment:
To be honest I don't feel  leaving it as 0 and 6 even if you used $dayIsWorkingDay still good and maybe it could be solved by creating another class that map days to constants since PHP don't support enumeration yet ...  but anyway I don't want to complicate things  I would say at least document that 0 = Sunday and Saturday = 6 with a link to php documentation http://php.net/manual/en/function.date.php   ...